src/encoded/static/components/collection.js

```diff
@@ -109,22 +141,58 @@ class FacetChart extends React.Component {
 
     render() {
         const { chartId, facet } = this.props;
+        let terms = [];
+        let otherTerms;
+
+        // If we have more facet terms than we allow to be displayed (determined through
+        // MAX_CHART_ITEMS), then just get MAX_CHART_ITEMS to display as usual, and collect the
+        // rest to display as "other."
+        if (facet.terms.length > MAX_CHART_ITEMS) {
+            // We have more facet terms than we display individually. First get the terms to
+            // display as individual items.
+            terms = facet.terms.slice(0, MAX_CHART_ITEMS);
+
+            // Now collect the terms to display as "other."
+            otherTerms = facet.terms.slice(MAX_CHART_ITEMS);
+        } else {
+            // We don't have enough facet terms for an "other" category, so just use the whole
+            // array of facet terms.
+            terms = facet.terms;
+            otherTerms = [];
+        }
 
         // Extract the arrays of labels from the facet keys, and the arrays of corresponding counts
         // from the facet doc_counts. Only use non-zero facet terms in the charts. IF we have no
         // usable data, both these arrays have no entries. componentDidMount assumes these arrays
         // have been populated.
         this.values = [];
         this.labels = [];
-        facet.terms.forEach((term) => {
+        terms.forEach((term) => {
             if (term.doc_count) {
                 this.values.push(term.doc_count);
                 this.labels.push(term.key);
             }
         });
 
+        // Add labels and values for "other" terms, if any. Add up all the "other" items and push
+        // as another data item to the array of values.
+        if (otherTerms.length) {
+            this.values.push(otherTerms.reduce((sum, term) => sum + term.doc_count, 0));
+            this.labels.push('other');
+
+            // Keep track of the "other" terms and its location (stored in this.otherTermsIndex) in
+            // the values/labels arrays so that the callbacks to handle clicks in the legend and
+            // the doughnut slices can handle the "other" term.
+            this.otherTerms = otherTerms;
+            this.otherTermsIndex = this.labels.length - 1;
+        } else {
+            // Remember we don't have enough terms to need an "other" term.
+            this.otherTerms = [];
+            this.otherTermsIndex = 0;
```

Comment:
Same as above, I know what's going on but it's a confusing code block since 'otherTermsIndex' is not an index for 'otherTerms'.  I'm not telling you to change your variable name but when I first saw this I thought, 'Why are we setting the index variable to 0 when the array is empty?'